qa/tasks/cephfs/test_nfs.py

```diff
@@ -145,11 +145,22 @@ def _test_create_cluster(self):
         '''
         Test single nfs cluster deployment.
         '''
-        # Disable any running nfs ganesha daemon
-        self._check_nfs_server_status()
-        self._nfs_cmd('cluster', 'create', self.cluster_id)
-        # Check for expected status and daemon name (nfs.<cluster_id>)
-        self._check_nfs_cluster_status('running', 'NFS Ganesha cluster deployment failed')
+        tries = 3
+        while True:
+            try:
+                # Disable any running nfs ganesha daemon
+                self._check_nfs_server_status()
+                self._nfs_cmd('cluster', 'create', self.cluster_id)
+                # Check for expected status and daemon name (nfs.<cluster_id>)
+                self._check_nfs_cluster_status(
+                    'running', 'NFS Ganesha cluster deployment failed')
+                break
+            except (AssertionError, CommandFailedError):
+                if tries:
+                    tries -= 1
+                    time.sleep(3)
+                    continue
+                raise
```

Comment:
I think a better method would be using `safe_while` and `MaxWhileTries` here. Checkout `test_mds_metrics.py`.

Reply:
thanks jos, will check them out

Reply:
i've pushed the change, PTAL